google_guest_agent_packaging_diff_tool.py

```diff
@@ -1,46 +1,101 @@
-from pprint import pprint
 import os
 import click
+import pprint
+import requests
+import warnings
+
 from debian import deb822
 
+UBUNTU_CONTROL_FILENAME = "ubuntu_control.file"
+GOOGLE_CONTROL_FILENAME = "gce_control.file"
+
+archive_dict = {
```

Comment:
it looks like the semantics of this dictionary are important. for readability, i'd recommend adding some typing here. E.g.

```
archives: dict[str, ArchiveMeta]

@dataclass
class ArchiveMeta:
    ubuntu_requirements: str
    google_requirements: str
```

and then corollary changes throughout the script

Reply:
ACK, good shout :smile: I'll add that now